File: 8 kyu/Sentence Smash/solution.ex
defmodule SentenceSmasher do
  def smash(words), do: Enum.join(words, " ")
end

_____________________________________
defmodule SentenceSmasher do
  def smash(words) do
    Enum.join(words, " ")
  end
end

_____________________________________
defmodule SentenceSmasher do
  def smash(words) do
    Enum.reduce(words, "", fn
      x, "" -> x
      x, acc -> acc <> " " <> x
    end)
  end
end

_____________________________________
defmodule SentenceSmasher do  
  def smash(words, acc \\ "")
  def smash([], acc), do: acc
  def smash([head | tail], ""), do: smash(tail, head) 
  def smash([head | tail], acc), do: smash(tail, "#{acc} #{head}")
end

File: 8 kyu/Sentence Smash/solution.rb
def smash(words)
  words.join ' '
end
_____________________________________
def smash(words)
  words * ' '
end

_____________________________________
def smash(words)
 return "" if words.empty?
  words.inject("") do |m,w|
      m << w << " "
  end.chop!
end

_____________________________________
def smash(words)
  words.join (' ')
end

words = Array.new
words << 'Mike'
words << 'Katya'

puts smash(words)
